hal/halx86/apic/apic.c

```diff
@@ -27,6 +27,12 @@
 ULONG ApicVersion;
 UCHAR HalpVectorToIndex[256];
 
+ULONG
+HalpGetCurrentProcessorHwID(VOID)
+{
+    return ApicRead(APIC_ID);
```

Comment:
The actual APIC Id is stored in the highest 8 bits. Is this intended to return the full field?